pages/CitationStyleLanguageHandler.inc.php

```diff
@@ -94,54 +97,45 @@ public function _setupRequest($args, $request) {
 
 		$this->citationStyle = $args[0];
 		$this->returnJson = isset($userVars['return']) && $userVars['return'] === 'json';
-		$this->article = Services::get('submission')->get($userVars['submissionId']);
-
-		if (!$this->article) {
+		$this->submission = Services::get('submission')->get($userVars['submissionId']);
+		$this->issue = $userVars['issueId'] ? Services::get('issue')->get($userVars['issueId']) : null;
+		$this->submissionNoun = $userVars['submissionNoun'];
+		
+		if (!$this->submission) {
 			$request->getDispatcher()->handle404();
 		}
-
-		$this->publication = !empty($userVars['publicationId'])
-			? Services::get('publication')->get($userVars['publicationId'])
-			: $this->article->getCurrentPublication();
-
-		if ($this->article) {
-			$issueDao = DAORegistry::getDAO('IssueDAO');
-			// Support OJS 3.1.x and 3.2
-			$issueId = method_exists($this->article, 'getCurrentPublication') ? $this->article->getCurrentPublication()->getData('issueId') : $this->article->getIssueId();
-			$this->issue = $issueDao->getById($issueId, $context->getId());
-		}
```

Comment:
This can go; in recent releases this plugin is managed as a submodule, and in general I'm not in favour of maintaining a single plugin codebase that's compatible with several branches of the software.